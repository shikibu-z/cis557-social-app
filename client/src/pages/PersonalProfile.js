/* eslint-disable no-unused-vars */
/* eslint-disable prefer-template */
import React, { useEffect, useState } from 'react';
import NavBar from '../components/NavBar';
import Profile from '../components/Profile';
import Editprofile from '../components/Editprofile';
import '../stylesheets/PersonalProfile.css';

import { fetchUserProfile, fetchUserGroups } from '../api/fetchers';

function PersonalProfile() {
  const id = localStorage.getItem('currUser');

  const [modalShow, setModalShow] = useState(false);
  const [profile, setProfile] = useState();
  const [groups, setGroups] = useState();
  const [isLoading, setIsLoading] = useState(true);

  function formatDate(d) {
    const m = (d.getMonth() + 1).toString();
    const day = d.getDate().toString();
    const y = d.getFullYear().toString();
    const res = m + '-' + day + '-' + y;
    return res;
  }

  useEffect(async () => {
    const prof = await fetchUserProfile(id);
    setProfile(prof.data);
    const userGroups = await fetchUserGroups(id);
    const gr = userGroups.data.map((g) => [g.profilePhoto, g.groupName]);
    setGroups(gr);
    setIsLoading(false);
  }, []);

  return (
    !isLoading
    && (
    <div>
      <NavBar isLogin={0} />
      <div className="container d-flex flex-column align-items-center justify-content-center">
        <Profile
          profileImage={profile.photo}
          userName={profile.username}
          registrationDate={formatDate(new Date(profile.registrationDate))}
          interests={profile.interests}
          gender={profile.gender}
          groupsInCommon={groups}
        />

        <div aria-hidden="true" id="edit-profile" onClick={() => setModalShow(true)}>
          <p style={{ color: 'white', 'font-size': '15px', 'font-weight': 'bold' }}>Edit Profile</p>
        </div>

        <Editprofile
          show={modalShow}
          onHide={() => setModalShow(false)}
          image={profile.photo}
        />
      </div>
    </div>
    )
  );
}

export default PersonalProfile;
