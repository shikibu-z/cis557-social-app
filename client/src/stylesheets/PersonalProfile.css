#edit-profile {
  width: 300px;
  height: 30px;
  position: absolute;
  top: 90%;
  
  background: #F8B114;
  border-radius: 8px;
  display: flex;
  align-items: center;
  justify-content: center;
}