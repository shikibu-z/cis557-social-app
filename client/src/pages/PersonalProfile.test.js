/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import PersonalProfile from './PersonalProfile';

test('Profile page matches snapshot', () => {
  const component = renderer.create(<PersonalProfile />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
