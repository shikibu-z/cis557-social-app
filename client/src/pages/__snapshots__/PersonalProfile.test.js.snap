// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page matches snapshot 1`] = `null`;
